google-ads-bom/build.gradle

```diff
@@ -0,0 +1,172 @@
+ *       <type>pom</type>
+ *       <scope>import</scope>
+ *     </dependency>
+ *   </dependencies>
+ * </dependencyManagement>
+ *
+ * Then, add dependencies on the Google Ads API library artifacts without
+ * specifying their versions.
+ */
+
+plugins {
+    id 'java-platform'
+    id 'maven-publish'
+    id 'org.jreleaser'
+}
+
+// Creates an empty JAR to be published alongside the BOM's POM file.
+// JReleaser requires at least one JAR to publish, and will not publish a
+// standalone POM file.
+task emptyJar(type: Jar) {
+    archiveClassifier = "empty"
+}
+
+// The code below is used to create a staging directory for jreleaser to deploy to.
+// This is required because the jreleaser task does not create the directory itself.
+task prepareJReleaser {
+    doLast {
+        mkdir "${buildDir}/staging-deploy"
+    }
+}
+
+// The jreleaser task depends on the prepareJReleaser task to ensure that the
+// staging directory is created before it runs.
+tasks.named("jreleaserConfig").configure {
+    dependsOn prepareJReleaser
+    dependsOn emptyJar
+    dependsOn generatePomFileForMavenPublication
+}
+
+jreleaser {
+    project {
+        gitRootSearch = true // Allow JReleaser to search upwards for .git
+        copyright = 'Google LLC'
+        description = 'Bill of Materials for the Google Ads API Client Library'
+        links {
+            homepage = 'https://github.com/googleads/google-ads-java'
+            documentation = 'https://developers.google.com/google-ads/api/docs/client-libs/java'
+        }
+        license = 'Apache-2.0'
+        authors = ['Josh Radcliff', 'Nick Birnie', 'Devin Chasanoff', 'Sarah Botwinick Pollack']
+    }
+    signing {
+        active = 'ALWAYS'
+        armored = true
+    }
+    deploy {
+        maven {
+            mavenCentral {
+                sonatype {
+                    active = 'ALWAYS'
+                    url = 'https://central.sonatype.com/api/v1/publisher'
+                    stagingRepository(layout.buildDirectory.dir("staging-deploy").get().toString())
+                    stage = 'UPLOAD'
+                }
+            }
+
+            release {
+                github {
+                    skipRelease = true
+                    skipTag = true
+                }
+            }
+        }
+    }
+}
+
+
+javaPlatform {
+    allowDependencies()
+}
+
+dependencies {
+    // This makes this BOM inherit all dependency versions from the Google Cloud
+    // BOM. This version is determined based on the current used version of
+    // google-cloud-shared-dependencies.
+    api(platform("com.google.cloud:libraries-bom:26.65.0"))
+
+    // Add constraints for all of this project's artifacts.
+    constraints {
+        api "com.google.api-ads:google-ads:${version}"
+        api "com.google.api-ads:google-ads-codegen:${version}"
+        api "com.google.api-ads:google-ads-shadowjar:${version}"
+        api "com.google.api-ads:google-ads-stubs-v19:${version}"
```

Comment:
WIll adding and removing versions from this file on major releases and sunset be automated?